V2 API Specification

Saoski opened the pull request:
## Issue
[Closes #67](https://github.com/rit-coms/COMS-Console/issues/67#issue-3088832235)

### Major Changes:
- No major changes to code base
- New Documentation for v1 and v2 APIs

### Setup / How to Test:
- Navigate to quackbox-api-docs\docs-v1\index.html (or the docs v2 folder) and you can view the HTML page (I used the Live Server VS Code extension)

### Testing Considerations:
- We need to verify that the v1 docs match the current implementation.
- We need to implement v2 routes to match the documentation.
- How can we display these pages? We should also add links to other versions of the API in the description section of the YAML file.

Saoski committed:
Add test api

Saoski committed:
Test docs

Saoski committed:
Update dependencies

Saoski committed:
Add dependency for asyncapi/html-template

Saoski committed:
remove dependencies

Saoski committed:
Revert "remove dependencies"

This reverts commit 101687f3e75f0bd46e014ab69ecbf285e5f8c082.

Saoski committed:
Add html template for asyncAPI docs

Saoski committed:
Add v1 API specification

Saoski committed:
Add HTML docs

Saoski committed:
Created v2 API doc

Saoski committed:
Delete example file

Saoski committed:
Fix error in v1 docs

Saoski committed:
Redefine leaderboard routes

Saoski committed:
Redefine save data routes

Saoski committed:
Fix typo

Saoski committed:
Fix message refernces

Saoski committed:
Generate new docs

Saoski committed:
Add WebSocket server

Saoski committed:
Add websocket operations

Saoski committed:
Fix typo

Saoski committed:
Generate new docs

Saoski committed:
Minor fixes

Saoski committed:
Add examples for messages

Saoski committed:
Minor changes

Saoski committed:
Generate new docs

Saoski committed:
Generate v1 docs

Saoski committed:
Reorganize docs folders

Matgenius04 committed:
Merge branch 'main' into dev/v2-api-specification

Matgenius04 reviewed (commented):
Overall, the documentation looks good! There are some super minor spelling mistakes I noticed.

I'm thinking of using GitHub Pages to host these, but it's a little bit tricky figuring out how to make everything work together nicely. I think if we're able to unify everything into one /docs folder from the root level directory with all the static assets for documentation, that'd be ideal.

NOTE: I have not actually validated whether the v1 or v2 APIs actually line up with what is described in the documentation.

Matgenius04 committed:
Adding doc-v1 and doc-v2 npm run commands for serving api docs

Saoski committed:
Fix spelling and license url errors

Saoski committed:
Change to LF endings

Saoski commented:
Before merging this, we need to host this documentation on GitHub pages.

Saoski committed:
Half fixed an operation (PLEASE FINISH)

Saoski committed:
Fix new save data schema

Saoski commented:
I have read over the v1 docs and they seem consistent with the code

Saoski committed:
Fix command for generating documentation

Saoski committed:
Merge remote-tracking branch 'origin/release/v2-dev' into dev/v2-api-specification

Saoski committed:
Update dependencies

Saoski committed:
Restructure api doc folder

Saoski committed:
Add workflow for uploading docs to github pages

Saoski committed:
Add workflow for uploading docs to github pages

Saoski committed:
Merge branch 'dev/v2-api-specification' of https://github.com/rit-coms/COMS-Console into dev/v2-api-specification

Saoski committed:
Fix path to async spec in workflow

Saoski committed:
Add force-write flag to generate docs in workflow

Saoski committed:
Attempt to fix workflow

Saoski committed:
Add permissions to github workflow

Saoski commented:
Documentation is on this site: https://rit-coms.github.io/COMS-Console/

Saoski commented:
I think the v2 routes should be implemented in a separate branch. This PR is ready for review.

Saoski committed:
Add a solution to an error for the asyncAPI generator in the README file

Saoski committed:
Modify GitHub Actions workflow for API docs

Updated workflow to trigger on 'main' and 'release/v2-dev' branches on push.

Saoski committed:
Remove GitHub Pages deployment step

Removed deployment step for GitHub Pages (This will be moved to another action).

Saoski committed:
Add workflow for publishing docs to Github pages
